tests/unit/scripting.tcl

```diff
@@ -257,16 +257,33 @@ start_server {tags {"scripting"}} {
         run_script {return redis.pcall('wait','1','0')} 0
     } {0}
 
-    test {EVAL - Scripts can't run XREAD and XREADGROUP with BLOCK option} {
+    test {EVAL - Scripts do not block on XREAD with BLOCK option} {
         r del s
         r xgroup create s g $ MKSTREAM
-        set res [run_script {return redis.pcall('xread','STREAMS','s','$')} 1 s]
+        set res [run_script {return redis.pcall('xread','STREAMS','s','$')} 0]
         assert {$res eq {}}
-        assert_error "*xread command is not allowed with BLOCK option from scripts" {run_script {return redis.pcall('xread','BLOCK',0,'STREAMS','s','$')} 1 s}
-        set res [run_script {return redis.pcall('xreadgroup','group','g','c','STREAMS','s','>')} 1 s]
+        run_script {return redis.pcall('xread','BLOCK',0,'STREAMS','s','$')} 0
+    } {}
+
+    test {EVAL - Scripts do not block on XREADGROUP with BLOCK option} {
+        set res [run_script {return redis.pcall('xreadgroup','group','g','c','STREAMS','s','>')} 0]
         assert {$res eq {}}
-        assert_error "*xreadgroup command is not allowed with BLOCK option from scripts" {run_script {return redis.pcall('xreadgroup','group','g','c','BLOCK',0,'STREAMS','s','>')} 1 s}
-    }
+        run_script {return redis.pcall('xreadgroup','group','g','c','BLOCK',0,'STREAMS','s','>')} 0
+    } {}
+
+    test {EVAL - Scripts do not block on XREAD with BLOCK option -- non empty stream} {
+        r XADD s * a 1
+        run_script {return redis.pcall('xread','BLOCK',0,'STREAMS','s','$')} 0
```

Comment:
i suppose it's better to use `0-0` here instead of `$`, and get some entry at output.
i.e. the above tests handle a case where there's nothing to return and the block is ignored returning an empty result. and these two tests below show that if there's data, the data is returned (block doesn't interferes with that).

Reply:
done